pkg/virt-launcher/virtwrap/agent-poller/agent_poller.go

```diff
@@ -588,10 +582,58 @@ func convertToUsers(guestInfo *libvirt.DomainGuestInfo) []api.User {
 	return users
 }
 
+func findDisk(guestInfoDisks []libvirt.DomainGuestInfoDisk, serial string) libvirt.DomainGuestInfoDisk {
+	for _, disk := range guestInfoDisks {
+		if disk.Serial == serial {
+			return disk
+		}
+	}
+	// Unreachable
+	return libvirt.DomainGuestInfoDisk{}
+}
```

Comment:
I think a better design here would be to return a pointer to the found disk, and `nil` if it is not found.
That eliminates copying, so has a little better performance, and handles the case when somehow the disk is not found.